mirascope/core/base/merge_decorators.py

```diff
@@ -0,0 +1,59 @@
+from collections.abc import Callable
+from functools import reduce, wraps
+from typing import Any, ParamSpec, TypeVar
+
+_P = ParamSpec("_P")
+_R = TypeVar("_R")
+
+_WP = ParamSpec("_WP")
+_WR = TypeVar("_WR")
+
+
+def merge_decorators(
+    decorator: Callable[[Callable[_P, _R]], Callable[_WP, _WR]],
+    *additional_decorators: Callable[[Callable], Callable],
+) -> Callable[[], Callable[[Callable[_P, _R]], Callable[_WP, _WR]]]:
```

Comment:
Due to current Python's type system limitations, it's challenging to apply precise type annotations to variable-length arguments. 

This compromise allows for proper type checking of both the wrapped and wrapper functions in merge_decorators' arguments, making it a pragmatic implementation. The first decorator determines the type constraints, while additional decorators use more flexible typing.